### PR TITLE
Add `operator+` for `Typedef`

### PR DESCRIPTION
**Pull request by akopich**
This adds `PlusTag`. The Typedefs of Tag that inherits from
`PlusTag` have `operator+` enabled.

Test: Unit
Bug: http://b/241077964

### CODE REVIEW

**Review by antonrohr — APPROVED** [review]
LGTM, but I am no expert on template things. Please wait for @beckerhe review

File: src/OrbitBase/include/OrbitBase/Typedef.h [hunk 1]
```diff
@@ -139,6 +139,18 @@ class Typedef {
...
+// When the `Tag` is inherits from the struct, the `Typedef<Tag, T>` supports `operator+`
+struct PlusTag {};
+
+template <typename First, typename Second, typename FirstDecayed = std::decay_t<First>,
+          typename SecondDecayed = std::decay_t<Second>, typename Tag = typename FirstDecayed::Tag,
+          typename = std::enable_if_t<
+              std::is_same_v<typename FirstDecayed::Tag, typename SecondDecayed::Tag>>,
+          typename = std::enable_if_t<std::is_base_of_v<PlusTag, Tag>>>
+auto operator+(First&& lhs, Second&& rhs) {
+  return Typedef<Tag, decltype(*lhs + *rhs)>(*lhs + *rhs);
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/Typedef.h:151]: This is missing std::forward

**Review by beckerhe — APPROVED** [review]
Nice. LGTM

File: src/OrbitBase/TypedefTest.cpp [hunk 2]
```diff
@@ -324,4 +325,34 @@ TEST(Typedef, ComparisonIsCorrect) {
...
+}
+
+TEST(Typedef, WrapperWithPlusHasPlusAndConvertsArgument) {
+  constexpr std::chrono::nanoseconds kNanos(1000);
+  constexpr std::chrono::microseconds kMicros(1);
+  WrapperWithPlus<std::chrono::nanoseconds> a(kNanos);
+  WrapperWithPlus<std::chrono::microseconds> b(kMicros);
+  EXPECT_EQ(*(a + b), kNanos + kMicros);
+}
+
```
Comment (beckerhe) [inline src/OrbitBase/TypedefTest.cpp:357]: A test for a move-only type would be great. This helps to test perfect forwarding even though it's not that likely to happen.

**Pull request merged**